authorization-server/src/modules/auth/auth.controller.ts

```diff
@@ -74,6 +81,17 @@ export class AuthController {
       )}`,
     );
 
+    return this.loginCommon(req, res);
+  }
+
+  /**
+   * implements logic common for all login methods
+   */
+  async loginCommon(req: Request, res: Response): Promise<LoginResponseDto> {
+    if (typeof req.user !== 'string') {
+      throw new Error(`unexpected req.user value: ${req.user}`);
```

Comment:
I decided to have an internal application error because req.user that is not a string here means something went really wrong and I've just checked that `decodeJWT` returns null when an empty string is provided. It does not throw an error in this case.